services/121-service/src/payments/fsp-integration/nedbank/nedbank-api.service.ts

```diff
@@ -0,0 +1,231 @@
+    }
+    const headers = this.createHeaders();
+
+    let response: AxiosResponse<T>;
+    try {
+      response = await this.httpService.request<AxiosResponse<T>>({
+        method,
+        url,
+        payload,
+        headers,
+        httpsAgent: this.httpsAgent,
+      });
+    } catch (error) {
+      console.error(`Failed to make Nedbank ${method} call`, error);
+      throw new NedbankError(`Error: ${error.message}`);
+    }
+    if (this.isNedbankErrorResponse(response.data)) {
+      const errorMessage = this.formatError(response.data);
+      throw new NedbankError(errorMessage);
+    }
+    return response;
+  }
+
+  private createHeaders(): Header[] {
+    return [
+      { name: 'x-ibm-client-id', value: process.env.NEDBANK_CLIENT_ID! },
+      {
+        name: 'x-ibm-client-secret',
+        value: process.env.NEDBANK_CLIENT_SECRET!,
+      },
+      {
+        name: 'x-idempotency-key', // We use OrderCreateReference as 'idempotency' key and therefore set this thing with a random value
+        value: Math.floor(Math.random() * 10000).toString(),
+      },
+      {
+        name: 'x-jws-signature',
+        value: Math.floor(Math.random() * 10000).toString(), // Should be a random integer https://apim.nedbank.co.za/static/docs/cashout-create-order
+      },
+      { name: 'x-fapi-financial-id', value: 'OB/2017/001' }, // Should always be this value https://apim.nedbank.co.za/static/docs/cashout-create-order
+      { name: 'x-fapi-customer-ip-address', value: process.env.PUBLIC_IP! },
+      { name: 'x-fapi-interaction-id', value: uuid() }, // Should be a UUID https://apim.nedbank.co.za/static/docs/cashout-create-order
+      { name: 'Content-Type', value: 'application/json' },
+    ];
+  }
+
+  private isNedbankErrorResponse(
+    response: unknown | ErrorReponseNedbankDto,
+  ): response is ErrorReponseNedbankDto {
+    return (response as ErrorReponseNedbankDto).Errors !== undefined;
+  }
+
+  private formatError(responseBody: ErrorReponseNedbankDto | null): string {
+    if (!responseBody) {
+      return 'Nebank URL could not be reached';
+    }
+    let errorMessage = '';
+
+    if (responseBody.Errors && responseBody.Errors.length > 0) {
+      const errorMessages = responseBody.Errors.map(
+        (error) => error.Message,
+      ).filter(Boolean);
+      errorMessage = `Errors: ${errorMessages.join('; ')}`;
+    }
+
+    const additionalInfo: string[] = [];
+    if (responseBody.Message) {
+      additionalInfo.push(`Message: ${responseBody.Code}`);
+    }
+
+    if (responseBody.Code) {
+      additionalInfo.push(`Code: ${responseBody.Code}`);
+    }
+    if (responseBody.Id) {
+      additionalInfo.push(`Id: ${responseBody.Id}`);
+    }
+
+    if (additionalInfo.length > 0) {
+      errorMessage += ` (${additionalInfo.join(', ')})`;
+    }
+
+    if (errorMessage === '') {
+      errorMessage = 'Unknown error';
+    }
+
+    return errorMessage;
+  }
+
+  private createHttpsAgent(): https.Agent | undefined {
```

Comment:
Did you consider unit testing this function(ality)?

Reply:
Consider testing this instead: createHttpsAgentWithCertificate